### PR TITLE
Bug fixes and refactoring

### PR DESCRIPTION
**Pull request by spyder-jeff**
When using expr, the ID de-duplication wasn't working because of the way logs were used to track state. Also fix a number of other issues causing oversized requests and too much work, leading to connection resets and problems parsing streaming response data. Error handling is improved and error messages are suppressed unless repetitive. These changes support customers with significantly higher data volumes than the past release.

```
$ make tests
go test -cover ./...
	spyderbat-event-forwarder/logwrapper		coverage: 0.0% of statements
	spyderbat-event-forwarder/record		coverage: 0.0% of statements
ok  	spyderbat-event-forwarder/api	0.004s	coverage: 38.1% of statements
ok  	spyderbat-event-forwarder/config	(cached)	coverage: 53.7% of statements
ok  	spyderbat-event-forwarder/lru	0.003s	coverage: 87.5% of statements
ok  	spyderbat-event-forwarder/lru/journal	(cached)	coverage: 83.9% of statements
ok  	spyderbat-event-forwarder/spyderbat-event-forwarder	0.861s	coverage: 28.6% of statements
ok  	spyderbat-event-forwarder/webhook	(cached)	coverage: 89.8% of statements
```

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -186,6 +190,11 @@ func (a *API) SourceDataQuery(ctx context.Context, st time.Time, et time.Time) (
 		return nil, err
 	}
 
+	/*
```
Comment (croteb) [inline api/api.go:193]: maybe bury this under a debug flag?

File: lru/journal/journal.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	}
+
+	return j, nil
+}
+
+// Add adds an entry to the journal.
+func (j *Journal) Add(id uint64) error {
+	if j.entries >= j.maxEntries {
+		// rotate the journal
+		err := j.rotate()
```
Comment (croteb) [inline lru/journal/journal.go:108]: dare i ask...does this need a lock around it?

Comment (spyder-jeff) [reply lru/journal/journal.go:108]: It doesn't

File: spyderbat-event-forwarder/forwarder.go [hunk 3]
```diff
@@ -43,57 +40,16 @@ var (
 const (
 	requestDelay    = 30 * time.Second // how long to wait between requests
 	minQueryOverlap = 5 * time.Minute  // always look back at least this far
+	maxQueryTime    = 15 * time.Minute // don't query more than this at once
```
Comment (croteb) [inline spyderbat-event-forwarder/forwarder.go:43]: part of me feels like we should turn this into a potential flag?  maybe in the future...

**Review by bc-sb — APPROVED** [review]

**Pull request merged**